dwave/optimization/src/nodes/collections.cpp

```diff
@@ -464,6 +464,47 @@ struct DisjointListStateData : NodeStateData {
         }
     }
 
+    // "Manually" set the full state of one of the disjoint lists. This will not check that
+    // the new state is a valid configuration (given the state of the other lists), so it
```

Comment:
Do we want to check valid when asserts are on? Feels like that will save some bugs in the future.

Reply:
This comment is slightly misleading as there is no way to check for correctness in this function. The function must be called multiple times on different lists to end up in a valid state if you are moving items between lists.

Reply:
I see. Maybe a validity check when `propagate()` is called then?